File: timesheet-leave-management-frontend/src/app/models/notification.ts
export interface Notification {
  id: string;
  type: string;
  title: string;
  message: string;
  isRead: boolean;
  isImportant: boolean;
  createdAtUtc: string;
  readAtUtc?: string;
  actionUrl?: string;
  entityType?: string;
  entityId?: string;
}

export interface NotificationListResponse {
  items: Notification[];
  page: number;
  pageSize: number;
  totalCount: number;
}

export interface NotificationPreference {
  id?: string;
  userId?: string;
  inAppEnabled: boolean;
  emailEnabled: boolean;
  updatedAtUtc?: string;
}

export interface NotificationTemplate {
  id: string;
  eventKey: string;
  name: string;
  channel?: string;
  subjectTemplate: string;
  bodyTemplate: string;
  description?: string | null;
  isActive: boolean;
  isCritical?: boolean;
  updatedAtUtc: string;
}

export interface UnreadCount {
  unreadCount: number;
}

File: TSLMS_Backend/NotificationService/Models/Notification.cs
using NotificationService.Enums;

namespace NotificationService.Models
{
    public class Notification
    {
        public Guid Id { get; set; }
        public Guid RecipientUserId { get; set; }
        public NotificationType Type { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Message { get; set; } = string.Empty;
        public bool IsRead { get; set; }
        public DateTime CreatedAtUtc { get; set; }
        public DateTime? ReadAtUtc { get; set; }
        public string? ActionUrl { get; set; }
        public string? EntityType { get; set; }
        public string? EntityId { get; set; }
        public bool IsImportant { get; set; }
    }
}
